File: 1168-DuplicateZeros/1168-DuplicateZeros.py
# Last updated: 4/29/2026, 10:25:40 AM
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        i = 0
        n = len(arr)
        while i < n:
            ch = arr[i]
            if ch == 0:
                arr.insert(i, 0)
                arr.pop()
                i += 1
            i += 1

File: 1168-DuplicateZeros/1168-DuplicateZeros.js
// Last updated: 4/29/2026, 10:25:34 AM
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    let i = 0;
    const n = arr.length;
    while(i < n) {
        if(arr[i] === 0) {
            arr.splice(i, 0, 0);
            arr.pop();
            i++;
        }
        i++;
    }
};